core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/stack/clusterproxy/reregister/ClusterProxyReRegistrationActions.java

```diff
@@ -34,28 +35,34 @@ public class ClusterProxyReRegistrationActions {
 
     @Bean(name = "CLUSTER_PROXY_RE_REGISTRATION_STATE")
     public Action<?, ?> reRegisterClusterProxyConfigAction() {
-        return new AbstractStackAction<ClusterProxyReRegistrationState, ClusterProxyReRegistrationEvent, StackContext, StackEvent>(StackEvent.class) {
+        return new AbstractStackAction<ClusterProxyReRegistrationState, ClusterProxyReRegistrationEvent,
+                                       ClusterProxyReRegistrationContext, ClusterProxyReRegistrationTriggerEvent>(ClusterProxyReRegistrationTriggerEvent.class) {
             @Override
-            protected StackContext createFlowContext(FlowParameters flowParameters,
-                StateContext<ClusterProxyReRegistrationState, ClusterProxyReRegistrationEvent> stateContext, StackEvent payload) {
-                return new StackContext(flowParameters, stackService.getById(payload.getResourceId()), null, null, null);
+            protected ClusterProxyReRegistrationContext createFlowContext(FlowParameters flowParameters,
+                    StateContext<ClusterProxyReRegistrationState, ClusterProxyReRegistrationEvent> stateContext,
+                    ClusterProxyReRegistrationTriggerEvent payload) {
+                return new ClusterProxyReRegistrationContext(flowParameters, stackService.getById(payload.getResourceId()), payload.getOriginalCrn());
             }
 
             @Override
-            protected void doExecute(StackContext context, StackEvent payload, Map<Object, Object> variables) {
+            protected void doExecute(ClusterProxyReRegistrationContext context, ClusterProxyReRegistrationTriggerEvent payload, Map<Object, Object> variables) {
                 sendEvent(context);
             }
 
             @Override
-            protected Selectable createRequest(StackContext context) {
+            protected Selectable createRequest(ClusterProxyReRegistrationContext context) {
+                if (context.getStack().getTunnel().useCcmV1()) {
+                    return new CCMV1RemapKeyRequest(context.getStack().getId(), context.getStack().getCloudPlatform(),
+                            CLUSTER_PROXY_RE_REGISTRATION_FINISHED_EVENT.event(), context.getOriginalCrn());
+                }
                 return new ClusterProxyReRegistrationRequest(
                         context.getStack().getId(), context.getStack().getCloudPlatform(),
                         CLUSTER_PROXY_RE_REGISTRATION_FINISHED_EVENT.event()
                 );
```

Comment:
As I understand the reregistration flow is triggered during the recovery phase. Don't we lose the old ssh key by they?

Reply:
Old key is stored under the new CRN so this will just switch it back.